<a href="https://colab.research.google.com/github/JarrydWannenburg/House_Prices_Adv_Reg/blob/main/0_House_Price_Pipeline.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Housing Prices - Advanced Regression Techniques

## Loading

In [ ]:
# Import libraries
import pandas as pd
import numpy as np
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer, SimpleImputer
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.compose import ColumnTransformer, make_column_selector as Selector, TransformedTargetRegressor
from sklearn.linear_model import LinearRegression, LassoCV, RidgeCV
from lightgbm import LGBMRegressor
from sklearn.ensemble import IsolationForest
from sklearn.preprocessing import StandardScaler, OneHotEncoder, OrdinalEncoder
from sklearn.base import BaseEstimator, TransformerMixin
from sklearn.pipeline import Pipeline, FeatureUnion
from sklearn.metrics import mean_squared_error, mean_absolute_error
from sklearn.decomposition import PCA
from sklearn.feature_selection import SelectKBest, f_regression, chi2
from catboost import CatBoostRegressor

In [ ]:
# Edit default display options since there are lots of features
pd.set_option('display.max_columns', None)

In [ ]:
df = pd.read_csv('./data/train.csv')

# Uncomment when running script to submit predictions
#df_test = pd.read_csv('./data/test.csv')

# Fix anomaly in the real test data
# print(df_test.loc[df_test['YrSold'] < df_test['YearBuilt']][['YearBuilt', 'YrSold']])
# df_test.loc[1089, 'YrSold'] = 2008
# df_test.loc[1089, 'YearBuilt'] = 2007
# df_test.loc[1089, ['YearBuilt', 'YrSold']]

      YearBuilt  YrSold
1089       2008    2007


YearBuilt    2007
YrSold       2008
Name: 1089, dtype: object

In [ ]:
df.shape

(1460, 81)

## Data check

In [ ]:
# Check for duplicate variables
print('training duplicate columns:', df.duplicated().sum())
print('testing duplicate columns:', df_test.duplicated().sum())

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

## Pipeline

In [ ]:
dict1 = {"MSSubClass": str, 'MoSold':str}#, 'YrSold':str}
df = df.astype(dict1)
df_test = df_test.astype(dict1)

In [ ]:
class CategoricalTransformer(BaseEstimator, TransformerMixin):
    def __init__(self, type_transformation_dict = {"MSSubClass": str,'MoSold':str}, 
                 excluded_cat_features=['PoolQC','Fence','MiscFeature', 'FireplaceQu']):
#         print('\nCat init \n')
        self.type_transformation_dict = type_transformation_dict
        self.excluded_cat_features = excluded_cat_features
        
    def fit(self, X, y=None):
#         print('\nCat fit \n')
        return self
    
    def transform(self, X, y=None):
#         print('\nCat trans \n')
        # Drop the excluded features (values with high null counts or useless)
        if len(self.excluded_cat_features) > 0:
            X.drop(columns=self.excluded_cat_features, inplace=True)
        # Transform types
        if len(self.type_transformation_dict) > 0:
            X.astype(self.type_transformation_dict)    

        X = X.replace([np.inf, -np.inf], np.nan)
        return X

In [ ]:
cat_pipe = Pipeline(steps=[
    ('transformer_function', CategoricalTransformer()),
    ('imputer', SimpleImputer(strategy='most_frequent')),
    ('encoder', OneHotEncoder(drop='if_binary', handle_unknown='ignore')),
    ('selectKbest', SelectKBest(k=200))
])

In [ ]:
class NumericTransformer(BaseEstimator, TransformerMixin):
    def __init__(self, right_skewed_cols=[], lower = 0.5, upper=10,
                excluded_num_features=['Id']):
#         print('\nNum init \n')
        self.right_skewed_cols = right_skewed_cols
        self.lower = lower
        self.upper = upper
        self.excluded_num_features = excluded_num_features
        
    def fit(self, X, y=None):
#         print('\nNum fit \n')
        return self
    
    def log_trans_right_skew(self, X, y=None):
#         print('\nNum log transform \n')
        # Get a list of columns with right skews
        self.right_skewed_cols = X.skew()[(X.skew() > lower) & (X.skew() < upper)].index.to_list()
        
        # Log transform right skew using log1p incase of 0 values
        for col in right_skewed_cols:
            X[col] = np.log1p(X[col])
            X.rename(columns={col:col+'_log'}, inplace=True)
        return X

    
    def transform(self, X, y=None):
        #print('\nNum trans \n')
        
        # Drop the excluded features (values with high null counts or useless)
        if len(self.excluded_num_features) > 0:
            X.drop(columns=self.excluded_num_features, inplace=True)
                   
        # Engineer new features and drop old ones to avoid multicollinearity
        X['bath_per_bed'] = (X['FullBath'] + 0.5*X['HalfBath'] + X['BsmtFullBath'] + X['BsmtHalfBath']) / X['BedroomAbvGr']
        #X['Bsmt_finished_prop'] = (X['BsmtFinSF1'] + X['BsmtFinSF2']) / X['TotalBsmtSF']
        X.drop(columns=['FullBath', 'HalfBath','BsmtFullBath','BsmtHalfBath'], inplace=True)
        
        # If data has right skew, log transform it to correct
        if len(self.right_skewed_cols) > 0:
            X = log_trans_right_skew(X)
            
        X = X.replace([np.inf, -np.inf], np.nan)
        return X

In [ ]:
num_pipe = Pipeline(steps=[
    ('transformer_function', NumericTransformer()),
    ('imputer', IterativeImputer()),
    ('scaler', StandardScaler()),
    ('pca', PCA(.85))
])

In [ ]:
preprocessor = ColumnTransformer(
    transformers=[
        ("cat", cat_pipe, Selector(dtype_exclude=np.number)),
        ("num", num_pipe, Selector(dtype_include=np.number))
    ]
)

pipe = Pipeline(steps=[
    ('preprocessor', preprocessor),
    ('model', LinearRegression()) #objective='RMSE', random_seed=0
])

In [ ]:
X, y = df.drop(columns=['SalePrice']), df.loc[:, 'SalePrice']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.3, random_state=101)

# Uncomment when running script for submission
# X_train,y_train,X_test = df.drop(columns=['SalePrice']),df.loc[:, ['SalePrice']], df_test.copy()

In [ ]:
pipe.fit(X,y)

C:\Users\jarry\anaconda3\lib\site-packages\sklearn\feature_selection\_univariate_selection.py:111: RuntimeWarning: divide by zero encountered in true_divide
  f = msb / msw


Pipeline(steps=[('preprocessor',
                 ColumnTransformer(transformers=[('cat',
                                                  Pipeline(steps=[('transformer_function',
                                                                   CategoricalTransformer()),
                                                                  ('imputer',
                                                                   SimpleImputer(strategy='most_frequent')),
                                                                  ('encoder',
                                                                   OneHotEncoder(drop='if_binary',
                                                                                 handle_unknown='ignore')),
                                                                  ('selectKbest',
                                                                   SelectKBest(k=200))]),
                                                  <sklearn.compose._column_transformer.make_column_selector object at 0x00000286941D9130>),
                                                 ('num',
                                                  Pipeline(steps=[('transformer_function',
                                                                   NumericTransformer()),
                                                                  ('imputer',
                                                                   IterativeImputer()),
                                                                  ('scaler',
                                                                   StandardScaler()),
                                                                  ('pca',
                                                                   PCA(n_components=0.85))]),
                                                  <sklearn.compose._column_transformer.make_column_selector object at 0x000002869462C250>)])),
                ('model', LinearRegression())])

In [ ]:
# Estimate SalePrice in log form since heavily skewed
class CustomTargetTransformer(BaseEstimator, TransformerMixin):
  # no need to implement __init__ in this particular case
    def fit(self, target):
        return self

    def transform(self, target):
        print('\n*****************target_transform() called.\n')
        target = np.log1p(target)
        return target

    def inverse_transform(self, target):
        print('\n*****************inverse_target_transform() called.\n')
        target = np.expm1(target)
        return target


# create a TargetTransformer 
model = TransformedTargetRegressor(regressor=pipe, 
                                   transformer=CustomTargetTransformer(), 
                                   check_inverse=False)

model.fit(X_train, y_train)
y_pred = model.predict(X_test)

pred = pd.DataFrame(y_pred, columns=['SalePrice'])
pred['Id'] = [i+1461 for i in range(0,len(pred))]
pred = pred[['Id','SalePrice']]
pred.head()

pred.to_csv('./submissions/Pipeline_log.csv', index=False)


*****************target_transform() called.

Learning rate set to 0.041084
0:	learn: 0.3844605	total: 2.48ms	remaining: 2.48s
1:	learn: 0.3738481	total: 4.69ms	remaining: 2.34s
2:	learn: 0.3642585	total: 6.95ms	remaining: 2.31s
3:	learn: 0.3552165	total: 9.12ms	remaining: 2.27s
4:	learn: 0.3464818	total: 11.3ms	remaining: 2.24s
5:	learn: 0.3372943	total: 13.3ms	remaining: 2.2s
6:	learn: 0.3283921	total: 15.5ms	remaining: 2.19s
7:	learn: 0.3207374	total: 17.5ms	remaining: 2.17s
8:	learn: 0.3131639	total: 19.6ms	remaining: 2.15s
9:	learn: 0.3053076	total: 21.7ms	remaining: 2.15s
10:	learn: 0.2982987	total: 23.7ms	remaining: 2.13s
11:	learn: 0.2916551	total: 25.8ms	remaining: 2.13s
12:	learn: 0.2846776	total: 27.9ms	remaining: 2.12s
13:	learn: 0.2779774	total: 29.9ms	remaining: 2.1s
14:	learn: 0.2722554	total: 32.2ms	remaining: 2.11s
15:	learn: 0.2661866	total: 34.3ms	remaining: 2.11s
16:	learn: 0.2606456	total: 36.6ms	remaining: 2.12s
17:	learn: 0.2556197	total: 38.6ms	remaining: 2.1s
1

183:	learn: 0.0988189	total: 355ms	remaining: 1.57s
184:	learn: 0.0986309	total: 357ms	remaining: 1.57s
185:	learn: 0.0984628	total: 359ms	remaining: 1.57s
186:	learn: 0.0982758	total: 360ms	remaining: 1.57s
187:	learn: 0.0981914	total: 362ms	remaining: 1.56s
188:	learn: 0.0980499	total: 364ms	remaining: 1.56s
189:	learn: 0.0978662	total: 366ms	remaining: 1.56s
190:	learn: 0.0977425	total: 368ms	remaining: 1.56s
191:	learn: 0.0976925	total: 370ms	remaining: 1.56s
192:	learn: 0.0974199	total: 372ms	remaining: 1.55s
193:	learn: 0.0971286	total: 374ms	remaining: 1.55s
194:	learn: 0.0970338	total: 376ms	remaining: 1.55s
195:	learn: 0.0968784	total: 377ms	remaining: 1.55s
196:	learn: 0.0966403	total: 380ms	remaining: 1.55s
197:	learn: 0.0964966	total: 381ms	remaining: 1.54s
198:	learn: 0.0964362	total: 383ms	remaining: 1.54s
199:	learn: 0.0961661	total: 385ms	remaining: 1.54s
200:	learn: 0.0960314	total: 387ms	remaining: 1.54s
201:	learn: 0.0958955	total: 389ms	remaining: 1.54s
202:	learn: 

381:	learn: 0.0706560	total: 712ms	remaining: 1.15s
382:	learn: 0.0705185	total: 714ms	remaining: 1.15s
383:	learn: 0.0703847	total: 716ms	remaining: 1.15s
384:	learn: 0.0703199	total: 718ms	remaining: 1.15s
385:	learn: 0.0701114	total: 720ms	remaining: 1.14s
386:	learn: 0.0700904	total: 722ms	remaining: 1.14s
387:	learn: 0.0699465	total: 723ms	remaining: 1.14s
388:	learn: 0.0697835	total: 725ms	remaining: 1.14s
389:	learn: 0.0696959	total: 727ms	remaining: 1.14s
390:	learn: 0.0696493	total: 729ms	remaining: 1.14s
391:	learn: 0.0695953	total: 731ms	remaining: 1.13s
392:	learn: 0.0694983	total: 733ms	remaining: 1.13s
393:	learn: 0.0694079	total: 735ms	remaining: 1.13s
394:	learn: 0.0692664	total: 736ms	remaining: 1.13s
395:	learn: 0.0692513	total: 738ms	remaining: 1.13s
396:	learn: 0.0691269	total: 740ms	remaining: 1.12s
397:	learn: 0.0690386	total: 742ms	remaining: 1.12s
398:	learn: 0.0688571	total: 743ms	remaining: 1.12s
399:	learn: 0.0686865	total: 745ms	remaining: 1.12s
400:	learn: 

582:	learn: 0.0524821	total: 1.07s	remaining: 766ms
583:	learn: 0.0524240	total: 1.07s	remaining: 765ms
584:	learn: 0.0523207	total: 1.07s	remaining: 763ms
585:	learn: 0.0522169	total: 1.08s	remaining: 761ms
586:	learn: 0.0521517	total: 1.08s	remaining: 759ms
587:	learn: 0.0520194	total: 1.08s	remaining: 757ms
588:	learn: 0.0518814	total: 1.08s	remaining: 755ms
589:	learn: 0.0517773	total: 1.08s	remaining: 753ms
590:	learn: 0.0516797	total: 1.09s	remaining: 752ms
591:	learn: 0.0515937	total: 1.09s	remaining: 750ms
592:	learn: 0.0514827	total: 1.09s	remaining: 748ms
593:	learn: 0.0514321	total: 1.09s	remaining: 746ms
594:	learn: 0.0513129	total: 1.09s	remaining: 744ms
595:	learn: 0.0512229	total: 1.09s	remaining: 742ms
596:	learn: 0.0512137	total: 1.1s	remaining: 741ms
597:	learn: 0.0511298	total: 1.1s	remaining: 739ms
598:	learn: 0.0511109	total: 1.1s	remaining: 737ms
599:	learn: 0.0510224	total: 1.1s	remaining: 735ms
600:	learn: 0.0509091	total: 1.1s	remaining: 733ms
601:	learn: 0.050

776:	learn: 0.0394608	total: 1.43s	remaining: 411ms
777:	learn: 0.0393974	total: 1.43s	remaining: 409ms
778:	learn: 0.0393093	total: 1.43s	remaining: 407ms
779:	learn: 0.0392172	total: 1.44s	remaining: 405ms
780:	learn: 0.0391546	total: 1.45s	remaining: 408ms
781:	learn: 0.0391130	total: 1.46s	remaining: 406ms
782:	learn: 0.0390314	total: 1.46s	remaining: 404ms
783:	learn: 0.0389814	total: 1.46s	remaining: 402ms
784:	learn: 0.0389208	total: 1.46s	remaining: 400ms
785:	learn: 0.0388545	total: 1.46s	remaining: 398ms
786:	learn: 0.0387889	total: 1.46s	remaining: 396ms
787:	learn: 0.0387332	total: 1.47s	remaining: 394ms
788:	learn: 0.0386321	total: 1.47s	remaining: 393ms
789:	learn: 0.0385864	total: 1.47s	remaining: 391ms
790:	learn: 0.0384861	total: 1.47s	remaining: 389ms
791:	learn: 0.0384396	total: 1.47s	remaining: 387ms
792:	learn: 0.0383735	total: 1.48s	remaining: 385ms
793:	learn: 0.0383679	total: 1.48s	remaining: 383ms
794:	learn: 0.0383082	total: 1.48s	remaining: 381ms
795:	learn: 

969:	learn: 0.0299181	total: 1.79s	remaining: 55.4ms
970:	learn: 0.0299045	total: 1.79s	remaining: 53.6ms
971:	learn: 0.0298278	total: 1.79s	remaining: 51.7ms
972:	learn: 0.0297967	total: 1.8s	remaining: 49.9ms
973:	learn: 0.0297561	total: 1.8s	remaining: 48ms
974:	learn: 0.0296980	total: 1.8s	remaining: 46.2ms
975:	learn: 0.0296658	total: 1.8s	remaining: 44.3ms
976:	learn: 0.0296182	total: 1.8s	remaining: 42.5ms
977:	learn: 0.0295681	total: 1.8s	remaining: 40.6ms
978:	learn: 0.0295174	total: 1.81s	remaining: 38.8ms
979:	learn: 0.0294719	total: 1.81s	remaining: 36.9ms
980:	learn: 0.0294402	total: 1.81s	remaining: 35.1ms
981:	learn: 0.0293850	total: 1.81s	remaining: 33.2ms
982:	learn: 0.0293420	total: 1.81s	remaining: 31.4ms
983:	learn: 0.0292933	total: 1.82s	remaining: 29.5ms
984:	learn: 0.0292478	total: 1.82s	remaining: 27.7ms
985:	learn: 0.0292141	total: 1.82s	remaining: 25.8ms
986:	learn: 0.0291598	total: 1.82s	remaining: 24ms
987:	learn: 0.0291524	total: 1.82s	remaining: 22.2ms
988

In [ ]:
mse = mean_squared_error(y_test, y_pred)
np.sqrt(mse)

27363.889117541577

## Hyper Param Tuning

In [ ]:
from catboost import CatBoostRegressor

class ClfSwitcher(BaseEstimator):
    def __init__(
        self, 
        estimator = LinearRegression()):
        """
        A Custom BaseEstimator that can switch between classifiers.
        :param estimator: sklearn object - The regressor/classifier
        """ 

        self.estimator = estimator


    def fit(self, X, y=None, **kwargs):
        self.estimator.fit(X, y)
        return self


    def predict(self, X, y=None):
        return self.estimator.predict(X)


    def predict_proba(self, X):
        return self.estimator.predict_proba(X)


    def score(self, X, y):
        return self.estimator.score(X, y)

### SelectKbest

In [ ]:
hyper_pipe = Pipeline(steps=[
    ('preprocessor', preprocessor)
])

full_pipe_clf = Pipeline([
    ('pipe', hyper_pipe),
    ('clf', ClfSwitcher())
])

parameters = [
    {
        'pipe__preprocessor__cat__selectKbest__k': [200],
        'pipe__preprocessor__num__pca__n_components': [0.85],
        'clf__estimator': [CatBoostRegressor(objective='RMSE')],
        #'clf__estimator__learning_rate': [0.01,0.015,0.02],
    }
]

import warnings
warnings.filterwarnings('ignore')

gscv = GridSearchCV(full_pipe_clf, parameters, cv=10)
gscv.fit(X, y)

Learning rate set to 0.042748
0:	learn: 78598.3695870	total: 2.4ms	remaining: 2.4s
1:	learn: 76477.2510250	total: 4.72ms	remaining: 2.35s
2:	learn: 74482.6765007	total: 7ms	remaining: 2.33s
3:	learn: 72562.0626991	total: 9.15ms	remaining: 2.28s
4:	learn: 70662.6692461	total: 11.3ms	remaining: 2.26s
5:	learn: 68776.7573099	total: 13.5ms	remaining: 2.23s
6:	learn: 67006.8046057	total: 15.5ms	remaining: 2.2s
7:	learn: 65562.1054582	total: 17.6ms	remaining: 2.18s
8:	learn: 63927.7106597	total: 20ms	remaining: 2.2s
9:	learn: 62275.7880923	total: 22.2ms	remaining: 2.2s
10:	learn: 60644.1049883	total: 24.5ms	remaining: 2.2s
11:	learn: 59232.1679982	total: 26.5ms	remaining: 2.18s
12:	learn: 57810.1206829	total: 28.8ms	remaining: 2.18s
13:	learn: 56530.8136190	total: 30.9ms	remaining: 2.17s
14:	learn: 55237.5556449	total: 32.9ms	remaining: 2.16s
15:	learn: 53949.9947409	total: 35ms	remaining: 2.15s
16:	learn: 52819.2492408	total: 37.3ms	remaining: 2.15s
17:	learn: 51746.0554296	total: 39.6ms	re

184:	learn: 18077.8827972	total: 357ms	remaining: 1.57s
185:	learn: 18045.1057143	total: 359ms	remaining: 1.57s
186:	learn: 17994.5017224	total: 361ms	remaining: 1.57s
187:	learn: 17957.1883807	total: 363ms	remaining: 1.57s
188:	learn: 17931.3531133	total: 364ms	remaining: 1.56s
189:	learn: 17873.2845023	total: 369ms	remaining: 1.57s
190:	learn: 17839.5388949	total: 372ms	remaining: 1.57s
191:	learn: 17792.4843562	total: 374ms	remaining: 1.57s
192:	learn: 17737.7684574	total: 377ms	remaining: 1.57s
193:	learn: 17696.4296979	total: 379ms	remaining: 1.57s
194:	learn: 17660.9247924	total: 381ms	remaining: 1.57s
195:	learn: 17619.3834625	total: 383ms	remaining: 1.57s
196:	learn: 17585.6583635	total: 385ms	remaining: 1.57s
197:	learn: 17542.7875152	total: 387ms	remaining: 1.57s
198:	learn: 17506.1202764	total: 389ms	remaining: 1.57s
199:	learn: 17483.1909985	total: 391ms	remaining: 1.56s
200:	learn: 17463.9811908	total: 393ms	remaining: 1.56s
201:	learn: 17429.3853282	total: 395ms	remaining

374:	learn: 12631.8076784	total: 716ms	remaining: 1.19s
375:	learn: 12589.5832565	total: 718ms	remaining: 1.19s
376:	learn: 12568.4208240	total: 720ms	remaining: 1.19s
377:	learn: 12546.6972441	total: 722ms	remaining: 1.19s
378:	learn: 12537.3153279	total: 724ms	remaining: 1.19s
379:	learn: 12511.4513842	total: 725ms	remaining: 1.18s
380:	learn: 12476.8614732	total: 727ms	remaining: 1.18s
381:	learn: 12457.6090425	total: 729ms	remaining: 1.18s
382:	learn: 12450.4856148	total: 731ms	remaining: 1.18s
383:	learn: 12419.6196503	total: 733ms	remaining: 1.18s
384:	learn: 12409.8927688	total: 735ms	remaining: 1.17s
385:	learn: 12397.4303901	total: 737ms	remaining: 1.17s
386:	learn: 12380.1477475	total: 739ms	remaining: 1.17s
387:	learn: 12370.6693239	total: 740ms	remaining: 1.17s
388:	learn: 12358.3339143	total: 742ms	remaining: 1.17s
389:	learn: 12335.5056759	total: 744ms	remaining: 1.16s
390:	learn: 12323.3767332	total: 746ms	remaining: 1.16s
391:	learn: 12309.2607433	total: 748ms	remaining

570:	learn: 9311.1689149	total: 1.08s	remaining: 809ms
571:	learn: 9305.0100924	total: 1.08s	remaining: 807ms
572:	learn: 9288.7519621	total: 1.08s	remaining: 805ms
573:	learn: 9269.2252904	total: 1.08s	remaining: 804ms
574:	learn: 9261.8068677	total: 1.08s	remaining: 802ms
575:	learn: 9244.4000248	total: 1.09s	remaining: 800ms
576:	learn: 9236.6330858	total: 1.09s	remaining: 798ms
577:	learn: 9226.0714473	total: 1.09s	remaining: 796ms
578:	learn: 9215.3144306	total: 1.09s	remaining: 794ms
579:	learn: 9194.2453024	total: 1.09s	remaining: 792ms
580:	learn: 9184.5503957	total: 1.09s	remaining: 790ms
581:	learn: 9167.6870933	total: 1.1s	remaining: 788ms
582:	learn: 9160.6565672	total: 1.1s	remaining: 786ms
583:	learn: 9146.6920896	total: 1.11s	remaining: 793ms
584:	learn: 9137.2850584	total: 1.11s	remaining: 791ms
585:	learn: 9132.3233255	total: 1.12s	remaining: 789ms
586:	learn: 9120.1476893	total: 1.12s	remaining: 787ms
587:	learn: 9104.6224277	total: 1.12s	remaining: 785ms
588:	learn: 

755:	learn: 7158.3614232	total: 1.44s	remaining: 464ms
756:	learn: 7155.5533342	total: 1.44s	remaining: 462ms
757:	learn: 7143.9514461	total: 1.44s	remaining: 460ms
758:	learn: 7134.4643335	total: 1.44s	remaining: 458ms
759:	learn: 7120.9626531	total: 1.44s	remaining: 456ms
760:	learn: 7116.9620096	total: 1.45s	remaining: 454ms
761:	learn: 7108.9332997	total: 1.45s	remaining: 452ms
762:	learn: 7098.6107306	total: 1.45s	remaining: 451ms
763:	learn: 7090.6371887	total: 1.45s	remaining: 449ms
764:	learn: 7078.1653910	total: 1.45s	remaining: 447ms
765:	learn: 7062.0194426	total: 1.46s	remaining: 445ms
766:	learn: 7049.5819832	total: 1.46s	remaining: 443ms
767:	learn: 7042.2472242	total: 1.46s	remaining: 441ms
768:	learn: 7030.0007302	total: 1.46s	remaining: 439ms
769:	learn: 7021.8026135	total: 1.46s	remaining: 437ms
770:	learn: 7015.7383599	total: 1.47s	remaining: 435ms
771:	learn: 7007.6975636	total: 1.47s	remaining: 433ms
772:	learn: 6993.0264381	total: 1.47s	remaining: 431ms
773:	learn

948:	learn: 5523.1535464	total: 1.8s	remaining: 96.6ms
949:	learn: 5514.0230312	total: 1.8s	remaining: 94.7ms
950:	learn: 5501.1031913	total: 1.8s	remaining: 92.8ms
951:	learn: 5491.8327231	total: 1.8s	remaining: 90.9ms
952:	learn: 5482.7619375	total: 1.8s	remaining: 89ms
953:	learn: 5477.5086711	total: 1.81s	remaining: 87.1ms
954:	learn: 5468.3907478	total: 1.81s	remaining: 85.2ms
955:	learn: 5460.0868257	total: 1.81s	remaining: 83.3ms
956:	learn: 5451.8856017	total: 1.81s	remaining: 81.5ms
957:	learn: 5443.7691273	total: 1.81s	remaining: 79.6ms
958:	learn: 5435.7428290	total: 1.82s	remaining: 77.7ms
959:	learn: 5429.3171390	total: 1.82s	remaining: 75.8ms
960:	learn: 5422.1803921	total: 1.82s	remaining: 73.9ms
961:	learn: 5413.2731772	total: 1.82s	remaining: 72ms
962:	learn: 5407.1465231	total: 1.82s	remaining: 70.1ms
963:	learn: 5399.4132089	total: 1.83s	remaining: 68.2ms
964:	learn: 5388.6308862	total: 1.83s	remaining: 66.3ms
965:	learn: 5379.7095452	total: 1.83s	remaining: 64.4ms
9

181:	learn: 17865.2567253	total: 352ms	remaining: 1.58s
182:	learn: 17830.0176203	total: 354ms	remaining: 1.58s
183:	learn: 17792.9846583	total: 356ms	remaining: 1.58s
184:	learn: 17749.3181993	total: 357ms	remaining: 1.57s
185:	learn: 17720.7382777	total: 359ms	remaining: 1.57s
186:	learn: 17701.7589791	total: 361ms	remaining: 1.57s
187:	learn: 17667.8933655	total: 363ms	remaining: 1.57s
188:	learn: 17633.3305900	total: 365ms	remaining: 1.57s
189:	learn: 17597.3625544	total: 367ms	remaining: 1.56s
190:	learn: 17529.9646224	total: 369ms	remaining: 1.56s
191:	learn: 17492.4150493	total: 371ms	remaining: 1.56s
192:	learn: 17472.4057472	total: 372ms	remaining: 1.56s
193:	learn: 17416.7725004	total: 374ms	remaining: 1.55s
194:	learn: 17378.9398737	total: 377ms	remaining: 1.55s
195:	learn: 17319.6114547	total: 378ms	remaining: 1.55s
196:	learn: 17286.4445828	total: 380ms	remaining: 1.55s
197:	learn: 17239.0878353	total: 382ms	remaining: 1.55s
198:	learn: 17199.8680599	total: 384ms	remaining

372:	learn: 12342.8237066	total: 711ms	remaining: 1.19s
373:	learn: 12315.1098197	total: 713ms	remaining: 1.19s
374:	learn: 12291.9328236	total: 714ms	remaining: 1.19s
375:	learn: 12261.9664399	total: 716ms	remaining: 1.19s
376:	learn: 12232.3815910	total: 718ms	remaining: 1.19s
377:	learn: 12200.4983486	total: 720ms	remaining: 1.18s
378:	learn: 12170.4387278	total: 722ms	remaining: 1.18s
379:	learn: 12149.4022050	total: 724ms	remaining: 1.18s
380:	learn: 12132.8668836	total: 726ms	remaining: 1.18s
381:	learn: 12100.2917186	total: 727ms	remaining: 1.18s
382:	learn: 12069.3991367	total: 729ms	remaining: 1.17s
383:	learn: 12040.9034615	total: 731ms	remaining: 1.17s
384:	learn: 12013.9116814	total: 733ms	remaining: 1.17s
385:	learn: 11991.3825511	total: 735ms	remaining: 1.17s
386:	learn: 11968.1629639	total: 737ms	remaining: 1.17s
387:	learn: 11949.6816409	total: 738ms	remaining: 1.16s
388:	learn: 11914.4610663	total: 740ms	remaining: 1.16s
389:	learn: 11881.8739227	total: 742ms	remaining

570:	learn: 8978.0496399	total: 1.07s	remaining: 804ms
571:	learn: 8963.5348430	total: 1.07s	remaining: 802ms
572:	learn: 8949.4484094	total: 1.07s	remaining: 800ms
573:	learn: 8937.5583156	total: 1.07s	remaining: 798ms
574:	learn: 8927.0821772	total: 1.08s	remaining: 796ms
575:	learn: 8916.6865548	total: 1.08s	remaining: 795ms
576:	learn: 8901.3218468	total: 1.08s	remaining: 793ms
577:	learn: 8890.8842014	total: 1.08s	remaining: 791ms
578:	learn: 8876.2991997	total: 1.08s	remaining: 789ms
579:	learn: 8859.5722940	total: 1.09s	remaining: 787ms
580:	learn: 8854.6085687	total: 1.09s	remaining: 785ms
581:	learn: 8836.1411148	total: 1.09s	remaining: 783ms
582:	learn: 8815.4503497	total: 1.09s	remaining: 781ms
583:	learn: 8800.7620698	total: 1.09s	remaining: 779ms
584:	learn: 8791.2044679	total: 1.1s	remaining: 778ms
585:	learn: 8780.7366708	total: 1.1s	remaining: 776ms
586:	learn: 8771.8756204	total: 1.1s	remaining: 774ms
587:	learn: 8753.8028119	total: 1.1s	remaining: 772ms
588:	learn: 87

768:	learn: 7015.0256449	total: 1.43s	remaining: 429ms
769:	learn: 7008.2110262	total: 1.43s	remaining: 428ms
770:	learn: 7002.9421637	total: 1.43s	remaining: 426ms
771:	learn: 6991.1894094	total: 1.44s	remaining: 424ms
772:	learn: 6982.2741206	total: 1.44s	remaining: 422ms
773:	learn: 6975.6973017	total: 1.44s	remaining: 420ms
774:	learn: 6969.6852461	total: 1.44s	remaining: 418ms
775:	learn: 6964.2902436	total: 1.44s	remaining: 416ms
776:	learn: 6958.7074645	total: 1.44s	remaining: 415ms
777:	learn: 6953.4424819	total: 1.45s	remaining: 413ms
778:	learn: 6951.0824828	total: 1.45s	remaining: 411ms
779:	learn: 6938.9780610	total: 1.45s	remaining: 409ms
780:	learn: 6930.8698834	total: 1.45s	remaining: 407ms
781:	learn: 6929.8734353	total: 1.45s	remaining: 405ms
782:	learn: 6924.3846014	total: 1.46s	remaining: 403ms
783:	learn: 6915.1041367	total: 1.46s	remaining: 402ms
784:	learn: 6909.7340434	total: 1.46s	remaining: 400ms
785:	learn: 6906.5708598	total: 1.46s	remaining: 398ms
786:	learn

963:	learn: 5641.8539430	total: 1.79s	remaining: 66.7ms
964:	learn: 5636.4358405	total: 1.79s	remaining: 64.9ms
965:	learn: 5625.7106552	total: 1.79s	remaining: 63ms
966:	learn: 5625.0311781	total: 1.79s	remaining: 61.2ms
967:	learn: 5615.0994792	total: 1.79s	remaining: 59.3ms
968:	learn: 5605.4641541	total: 1.8s	remaining: 57.5ms
969:	learn: 5598.7191777	total: 1.8s	remaining: 55.6ms
970:	learn: 5588.7065209	total: 1.8s	remaining: 53.8ms
971:	learn: 5582.0387156	total: 1.8s	remaining: 51.9ms
972:	learn: 5572.9473488	total: 1.8s	remaining: 50ms
973:	learn: 5567.0055270	total: 1.8s	remaining: 48.2ms
974:	learn: 5562.9840957	total: 1.81s	remaining: 46.3ms
975:	learn: 5553.2665831	total: 1.81s	remaining: 44.5ms
976:	learn: 5539.6304850	total: 1.81s	remaining: 42.6ms
977:	learn: 5532.8599128	total: 1.81s	remaining: 40.8ms
978:	learn: 5529.7700370	total: 1.81s	remaining: 38.9ms
979:	learn: 5524.1728244	total: 1.82s	remaining: 37.1ms
980:	learn: 5518.3739457	total: 1.82s	remaining: 35.2ms
98

162:	learn: 18760.7817711	total: 353ms	remaining: 1.81s
163:	learn: 18706.8802821	total: 355ms	remaining: 1.81s
164:	learn: 18660.9619365	total: 357ms	remaining: 1.81s
165:	learn: 18606.2710604	total: 359ms	remaining: 1.8s
166:	learn: 18552.1324749	total: 361ms	remaining: 1.8s
167:	learn: 18503.2447065	total: 363ms	remaining: 1.79s
168:	learn: 18460.1174498	total: 365ms	remaining: 1.79s
169:	learn: 18432.3662422	total: 367ms	remaining: 1.79s
170:	learn: 18372.4168907	total: 369ms	remaining: 1.79s
171:	learn: 18324.9033575	total: 371ms	remaining: 1.78s
172:	learn: 18294.5049664	total: 373ms	remaining: 1.78s
173:	learn: 18256.2076664	total: 374ms	remaining: 1.78s
174:	learn: 18212.3962329	total: 376ms	remaining: 1.77s
175:	learn: 18166.5054789	total: 378ms	remaining: 1.77s
176:	learn: 18126.4523666	total: 380ms	remaining: 1.77s
177:	learn: 18092.7283155	total: 382ms	remaining: 1.76s
178:	learn: 18050.9267915	total: 384ms	remaining: 1.76s
179:	learn: 17996.1037694	total: 386ms	remaining: 

354:	learn: 12867.0216789	total: 710ms	remaining: 1.29s
355:	learn: 12839.3364336	total: 712ms	remaining: 1.29s
356:	learn: 12801.8699176	total: 714ms	remaining: 1.29s
357:	learn: 12791.4772473	total: 716ms	remaining: 1.28s
358:	learn: 12776.9595256	total: 718ms	remaining: 1.28s
359:	learn: 12751.9187979	total: 720ms	remaining: 1.28s
360:	learn: 12732.1888693	total: 722ms	remaining: 1.28s
361:	learn: 12702.9614074	total: 724ms	remaining: 1.27s
362:	learn: 12671.8135067	total: 725ms	remaining: 1.27s
363:	learn: 12654.6419874	total: 727ms	remaining: 1.27s
364:	learn: 12636.4203339	total: 729ms	remaining: 1.27s
365:	learn: 12617.0146190	total: 731ms	remaining: 1.27s
366:	learn: 12599.9708188	total: 733ms	remaining: 1.26s
367:	learn: 12576.2198247	total: 735ms	remaining: 1.26s
368:	learn: 12561.0515497	total: 736ms	remaining: 1.26s
369:	learn: 12535.4853863	total: 738ms	remaining: 1.26s
370:	learn: 12513.7441129	total: 740ms	remaining: 1.25s
371:	learn: 12490.1715998	total: 742ms	remaining

550:	learn: 9706.6296792	total: 1.07s	remaining: 871ms
551:	learn: 9695.9116115	total: 1.07s	remaining: 869ms
552:	learn: 9685.7026446	total: 1.07s	remaining: 867ms
553:	learn: 9670.1185274	total: 1.07s	remaining: 865ms
554:	learn: 9668.1203933	total: 1.08s	remaining: 863ms
555:	learn: 9656.1172143	total: 1.08s	remaining: 861ms
556:	learn: 9638.1707901	total: 1.08s	remaining: 859ms
557:	learn: 9627.2831364	total: 1.08s	remaining: 857ms
558:	learn: 9605.3451152	total: 1.08s	remaining: 855ms
559:	learn: 9586.9472493	total: 1.08s	remaining: 853ms
560:	learn: 9579.5888038	total: 1.09s	remaining: 851ms
561:	learn: 9563.4862916	total: 1.09s	remaining: 849ms
562:	learn: 9556.7608751	total: 1.09s	remaining: 847ms
563:	learn: 9543.7951806	total: 1.09s	remaining: 845ms
564:	learn: 9526.6273914	total: 1.09s	remaining: 843ms
565:	learn: 9507.9933121	total: 1.1s	remaining: 842ms
566:	learn: 9503.3691091	total: 1.1s	remaining: 839ms
567:	learn: 9493.7875261	total: 1.1s	remaining: 837ms
568:	learn: 9

747:	learn: 7352.7389778	total: 1.43s	remaining: 482ms
748:	learn: 7343.0350386	total: 1.43s	remaining: 480ms
749:	learn: 7333.4274228	total: 1.43s	remaining: 478ms
750:	learn: 7319.5242526	total: 1.43s	remaining: 476ms
751:	learn: 7312.9499064	total: 1.44s	remaining: 474ms
752:	learn: 7303.1128696	total: 1.44s	remaining: 472ms
753:	learn: 7291.4166850	total: 1.44s	remaining: 470ms
754:	learn: 7274.0566701	total: 1.44s	remaining: 468ms
755:	learn: 7270.4986896	total: 1.44s	remaining: 466ms
756:	learn: 7261.5507147	total: 1.45s	remaining: 464ms
757:	learn: 7254.5363529	total: 1.45s	remaining: 462ms
758:	learn: 7245.0408183	total: 1.45s	remaining: 460ms
759:	learn: 7240.1105528	total: 1.45s	remaining: 458ms
760:	learn: 7230.7295809	total: 1.45s	remaining: 456ms
761:	learn: 7221.2865048	total: 1.46s	remaining: 455ms
762:	learn: 7209.8057955	total: 1.46s	remaining: 453ms
763:	learn: 7198.4086329	total: 1.46s	remaining: 451ms
764:	learn: 7190.8343109	total: 1.46s	remaining: 449ms
765:	learn

943:	learn: 5794.0066957	total: 1.79s	remaining: 106ms
944:	learn: 5790.4473054	total: 1.79s	remaining: 104ms
945:	learn: 5787.7933309	total: 1.79s	remaining: 102ms
946:	learn: 5777.7080806	total: 1.79s	remaining: 100ms
947:	learn: 5775.3164877	total: 1.79s	remaining: 98.5ms
948:	learn: 5768.2128520	total: 1.8s	remaining: 96.6ms
949:	learn: 5759.2180692	total: 1.8s	remaining: 94.7ms
950:	learn: 5749.9482368	total: 1.8s	remaining: 92.8ms
951:	learn: 5739.0875753	total: 1.8s	remaining: 90.9ms
952:	learn: 5732.0037085	total: 1.8s	remaining: 89ms
953:	learn: 5725.6408912	total: 1.81s	remaining: 87.1ms
954:	learn: 5718.5641689	total: 1.81s	remaining: 85.2ms
955:	learn: 5707.9980809	total: 1.81s	remaining: 83.3ms
956:	learn: 5702.9305971	total: 1.81s	remaining: 81.4ms
957:	learn: 5700.7676553	total: 1.81s	remaining: 79.5ms
958:	learn: 5692.9246463	total: 1.81s	remaining: 77.6ms
959:	learn: 5682.8232112	total: 1.82s	remaining: 75.7ms
960:	learn: 5677.5938831	total: 1.82s	remaining: 73.8ms
961

170:	learn: 18059.0298789	total: 347ms	remaining: 1.68s
171:	learn: 18036.5163666	total: 349ms	remaining: 1.68s
172:	learn: 17988.5730406	total: 351ms	remaining: 1.68s
173:	learn: 17935.8125997	total: 353ms	remaining: 1.68s
174:	learn: 17905.9765730	total: 355ms	remaining: 1.67s
175:	learn: 17860.9900139	total: 357ms	remaining: 1.67s
176:	learn: 17812.3818671	total: 359ms	remaining: 1.67s
177:	learn: 17785.9044972	total: 361ms	remaining: 1.67s
178:	learn: 17754.7107821	total: 363ms	remaining: 1.67s
179:	learn: 17712.5526812	total: 365ms	remaining: 1.66s
180:	learn: 17651.4215851	total: 367ms	remaining: 1.66s
181:	learn: 17604.5026125	total: 369ms	remaining: 1.66s
182:	learn: 17561.8273597	total: 371ms	remaining: 1.66s
183:	learn: 17525.9140511	total: 373ms	remaining: 1.65s
184:	learn: 17490.2595401	total: 375ms	remaining: 1.65s
185:	learn: 17469.4344897	total: 376ms	remaining: 1.65s
186:	learn: 17422.2531582	total: 378ms	remaining: 1.64s
187:	learn: 17377.2798610	total: 380ms	remaining

360:	learn: 12582.3349289	total: 706ms	remaining: 1.25s
361:	learn: 12559.2026425	total: 708ms	remaining: 1.25s
362:	learn: 12536.4006449	total: 710ms	remaining: 1.25s
363:	learn: 12523.7555716	total: 711ms	remaining: 1.24s
364:	learn: 12510.2966102	total: 714ms	remaining: 1.24s
365:	learn: 12503.1463311	total: 715ms	remaining: 1.24s
366:	learn: 12474.4347377	total: 718ms	remaining: 1.24s
367:	learn: 12450.0490453	total: 719ms	remaining: 1.24s
368:	learn: 12423.0274322	total: 721ms	remaining: 1.23s
369:	learn: 12404.6983063	total: 723ms	remaining: 1.23s
370:	learn: 12386.4251532	total: 725ms	remaining: 1.23s
371:	learn: 12366.4155971	total: 727ms	remaining: 1.23s
372:	learn: 12353.5667537	total: 729ms	remaining: 1.23s
373:	learn: 12335.8660645	total: 731ms	remaining: 1.22s
374:	learn: 12316.6238312	total: 733ms	remaining: 1.22s
375:	learn: 12294.2203435	total: 735ms	remaining: 1.22s
376:	learn: 12274.1832410	total: 737ms	remaining: 1.22s
377:	learn: 12257.1563496	total: 739ms	remaining

553:	learn: 9376.5040180	total: 1.06s	remaining: 857ms
554:	learn: 9367.2726530	total: 1.06s	remaining: 855ms
555:	learn: 9348.0735245	total: 1.07s	remaining: 853ms
556:	learn: 9326.5239705	total: 1.07s	remaining: 851ms
557:	learn: 9311.1586252	total: 1.07s	remaining: 849ms
558:	learn: 9295.4092916	total: 1.07s	remaining: 847ms
559:	learn: 9284.6631422	total: 1.07s	remaining: 845ms
560:	learn: 9270.8355189	total: 1.08s	remaining: 843ms
561:	learn: 9257.1481047	total: 1.08s	remaining: 841ms
562:	learn: 9234.6480415	total: 1.08s	remaining: 839ms
563:	learn: 9216.4781147	total: 1.08s	remaining: 837ms
564:	learn: 9212.3300740	total: 1.08s	remaining: 835ms
565:	learn: 9199.8863978	total: 1.09s	remaining: 833ms
566:	learn: 9188.9637099	total: 1.09s	remaining: 831ms
567:	learn: 9170.8554986	total: 1.09s	remaining: 829ms
568:	learn: 9152.4955595	total: 1.09s	remaining: 827ms
569:	learn: 9141.1064041	total: 1.09s	remaining: 825ms
570:	learn: 9132.5498049	total: 1.09s	remaining: 823ms
571:	learn

749:	learn: 7314.7097340	total: 1.42s	remaining: 474ms
750:	learn: 7308.0591421	total: 1.42s	remaining: 472ms
751:	learn: 7298.9315154	total: 1.43s	remaining: 471ms
752:	learn: 7284.3362508	total: 1.43s	remaining: 469ms
753:	learn: 7275.6162552	total: 1.43s	remaining: 467ms
754:	learn: 7261.3813449	total: 1.43s	remaining: 465ms
755:	learn: 7245.0021844	total: 1.43s	remaining: 463ms
756:	learn: 7238.5802865	total: 1.44s	remaining: 461ms
757:	learn: 7232.5206238	total: 1.44s	remaining: 459ms
758:	learn: 7223.1354193	total: 1.44s	remaining: 457ms
759:	learn: 7212.9095293	total: 1.44s	remaining: 456ms
760:	learn: 7197.9687040	total: 1.44s	remaining: 454ms
761:	learn: 7183.9665941	total: 1.45s	remaining: 452ms
762:	learn: 7183.3907235	total: 1.45s	remaining: 450ms
763:	learn: 7173.5449415	total: 1.45s	remaining: 448ms
764:	learn: 7165.8666427	total: 1.45s	remaining: 446ms
765:	learn: 7161.0734103	total: 1.45s	remaining: 444ms
766:	learn: 7154.0028665	total: 1.46s	remaining: 442ms
767:	learn

944:	learn: 5769.3418648	total: 1.78s	remaining: 104ms
945:	learn: 5766.5409491	total: 1.78s	remaining: 102ms
946:	learn: 5755.0751208	total: 1.79s	remaining: 100ms
947:	learn: 5745.8671755	total: 1.79s	remaining: 98.1ms
948:	learn: 5736.4922116	total: 1.79s	remaining: 96.2ms
949:	learn: 5734.5486518	total: 1.79s	remaining: 94.3ms
950:	learn: 5729.0184537	total: 1.79s	remaining: 92.4ms
951:	learn: 5722.5032149	total: 1.79s	remaining: 90.5ms
952:	learn: 5711.6627344	total: 1.8s	remaining: 88.7ms
953:	learn: 5702.1640132	total: 1.8s	remaining: 86.8ms
954:	learn: 5693.8493154	total: 1.8s	remaining: 84.9ms
955:	learn: 5684.8409204	total: 1.8s	remaining: 83ms
956:	learn: 5678.7026174	total: 1.8s	remaining: 81.1ms
957:	learn: 5670.5737584	total: 1.81s	remaining: 79.2ms
958:	learn: 5659.5748804	total: 1.81s	remaining: 77.3ms
959:	learn: 5654.2281673	total: 1.81s	remaining: 75.5ms
960:	learn: 5643.2977337	total: 1.81s	remaining: 73.6ms
961:	learn: 5636.9576660	total: 1.81s	remaining: 71.7ms
96

179:	learn: 17378.8566778	total: 347ms	remaining: 1.58s
180:	learn: 17339.1662205	total: 349ms	remaining: 1.58s
181:	learn: 17279.3397695	total: 351ms	remaining: 1.58s
182:	learn: 17227.4877498	total: 353ms	remaining: 1.57s
183:	learn: 17197.8091312	total: 354ms	remaining: 1.57s
184:	learn: 17156.6828870	total: 356ms	remaining: 1.57s
185:	learn: 17096.7267932	total: 358ms	remaining: 1.57s
186:	learn: 17059.8850928	total: 360ms	remaining: 1.56s
187:	learn: 17016.0092524	total: 362ms	remaining: 1.56s
188:	learn: 16985.0190741	total: 364ms	remaining: 1.56s
189:	learn: 16920.6239063	total: 366ms	remaining: 1.56s
190:	learn: 16892.3178358	total: 368ms	remaining: 1.56s
191:	learn: 16849.7707761	total: 369ms	remaining: 1.55s
192:	learn: 16797.2534315	total: 371ms	remaining: 1.55s
193:	learn: 16761.9408463	total: 373ms	remaining: 1.55s
194:	learn: 16728.1117519	total: 375ms	remaining: 1.55s
195:	learn: 16670.0533663	total: 377ms	remaining: 1.54s
196:	learn: 16628.3118775	total: 378ms	remaining

377:	learn: 11954.0407185	total: 704ms	remaining: 1.16s
378:	learn: 11939.6066383	total: 706ms	remaining: 1.16s
379:	learn: 11921.2969279	total: 708ms	remaining: 1.16s
380:	learn: 11910.7842802	total: 710ms	remaining: 1.15s
381:	learn: 11900.9125760	total: 712ms	remaining: 1.15s
382:	learn: 11887.7972801	total: 714ms	remaining: 1.15s
383:	learn: 11874.7187722	total: 715ms	remaining: 1.15s
384:	learn: 11851.9613075	total: 717ms	remaining: 1.15s
385:	learn: 11830.1149830	total: 719ms	remaining: 1.14s
386:	learn: 11811.7726462	total: 721ms	remaining: 1.14s
387:	learn: 11797.3478174	total: 723ms	remaining: 1.14s
388:	learn: 11777.7458006	total: 725ms	remaining: 1.14s
389:	learn: 11761.2949093	total: 726ms	remaining: 1.14s
390:	learn: 11751.4620621	total: 728ms	remaining: 1.13s
391:	learn: 11736.8700959	total: 730ms	remaining: 1.13s
392:	learn: 11725.9933085	total: 732ms	remaining: 1.13s
393:	learn: 11704.1273508	total: 734ms	remaining: 1.13s
394:	learn: 11675.4648118	total: 736ms	remaining

576:	learn: 8807.6851703	total: 1.06s	remaining: 780ms
577:	learn: 8792.5288112	total: 1.07s	remaining: 778ms
578:	learn: 8782.7982439	total: 1.07s	remaining: 777ms
579:	learn: 8770.0301189	total: 1.07s	remaining: 775ms
580:	learn: 8761.6198268	total: 1.07s	remaining: 773ms
581:	learn: 8756.1113327	total: 1.07s	remaining: 771ms
582:	learn: 8742.5441631	total: 1.07s	remaining: 769ms
583:	learn: 8728.4985471	total: 1.08s	remaining: 768ms
584:	learn: 8717.3185412	total: 1.08s	remaining: 766ms
585:	learn: 8709.9442583	total: 1.08s	remaining: 764ms
586:	learn: 8691.7699528	total: 1.08s	remaining: 762ms
587:	learn: 8677.9285819	total: 1.08s	remaining: 760ms
588:	learn: 8662.0252734	total: 1.09s	remaining: 758ms
589:	learn: 8656.0704209	total: 1.09s	remaining: 757ms
590:	learn: 8639.0809589	total: 1.09s	remaining: 755ms
591:	learn: 8616.6039844	total: 1.09s	remaining: 753ms
592:	learn: 8599.7059573	total: 1.09s	remaining: 751ms
593:	learn: 8588.3599651	total: 1.1s	remaining: 749ms
594:	learn:

773:	learn: 6800.2541801	total: 1.42s	remaining: 415ms
774:	learn: 6794.0336527	total: 1.42s	remaining: 413ms
775:	learn: 6786.0493807	total: 1.43s	remaining: 412ms
776:	learn: 6776.4879707	total: 1.43s	remaining: 410ms
777:	learn: 6769.6535747	total: 1.43s	remaining: 408ms
778:	learn: 6754.7293731	total: 1.43s	remaining: 406ms
779:	learn: 6743.6380302	total: 1.43s	remaining: 404ms
780:	learn: 6738.6109878	total: 1.44s	remaining: 402ms
781:	learn: 6729.7264468	total: 1.44s	remaining: 401ms
782:	learn: 6720.1440073	total: 1.44s	remaining: 399ms
783:	learn: 6718.5258747	total: 1.44s	remaining: 397ms
784:	learn: 6710.1508769	total: 1.44s	remaining: 395ms
785:	learn: 6698.0763162	total: 1.44s	remaining: 393ms
786:	learn: 6687.8576695	total: 1.45s	remaining: 391ms
787:	learn: 6679.3316084	total: 1.45s	remaining: 390ms
788:	learn: 6671.5583177	total: 1.45s	remaining: 388ms
789:	learn: 6662.2076780	total: 1.45s	remaining: 386ms
790:	learn: 6644.4776542	total: 1.45s	remaining: 384ms
791:	learn

971:	learn: 5363.2541993	total: 1.78s	remaining: 51.3ms
972:	learn: 5357.4545538	total: 1.78s	remaining: 49.5ms
973:	learn: 5352.1841306	total: 1.78s	remaining: 47.7ms
974:	learn: 5346.1234069	total: 1.79s	remaining: 45.8ms
975:	learn: 5341.2115075	total: 1.79s	remaining: 44ms
976:	learn: 5337.9446782	total: 1.79s	remaining: 42.2ms
977:	learn: 5330.0884060	total: 1.79s	remaining: 40.3ms
978:	learn: 5323.9592940	total: 1.79s	remaining: 38.5ms
979:	learn: 5319.4942839	total: 1.8s	remaining: 36.7ms
980:	learn: 5314.4449095	total: 1.8s	remaining: 34.8ms
981:	learn: 5306.8579775	total: 1.8s	remaining: 33ms
982:	learn: 5299.8087954	total: 1.8s	remaining: 31.2ms
983:	learn: 5291.5068048	total: 1.8s	remaining: 29.3ms
984:	learn: 5285.3801090	total: 1.8s	remaining: 27.5ms
985:	learn: 5280.4744449	total: 1.81s	remaining: 25.7ms
986:	learn: 5272.6666602	total: 1.81s	remaining: 23.8ms
987:	learn: 5264.1553364	total: 1.81s	remaining: 22ms
988:	learn: 5261.0015616	total: 1.81s	remaining: 20.2ms
989:

177:	learn: 18304.8179543	total: 348ms	remaining: 1.6s
178:	learn: 18259.4931258	total: 349ms	remaining: 1.6s
179:	learn: 18190.6328484	total: 351ms	remaining: 1.6s
180:	learn: 18137.5518797	total: 353ms	remaining: 1.6s
181:	learn: 18107.6330756	total: 355ms	remaining: 1.59s
182:	learn: 18050.0230153	total: 357ms	remaining: 1.59s
183:	learn: 18005.8326735	total: 359ms	remaining: 1.59s
184:	learn: 17942.3116010	total: 361ms	remaining: 1.59s
185:	learn: 17893.2270495	total: 363ms	remaining: 1.59s
186:	learn: 17858.9319188	total: 365ms	remaining: 1.59s
187:	learn: 17809.9569768	total: 367ms	remaining: 1.58s
188:	learn: 17758.6783571	total: 369ms	remaining: 1.58s
189:	learn: 17712.3811420	total: 370ms	remaining: 1.58s
190:	learn: 17694.3457127	total: 372ms	remaining: 1.58s
191:	learn: 17662.6128787	total: 374ms	remaining: 1.57s
192:	learn: 17626.9854702	total: 376ms	remaining: 1.57s
193:	learn: 17579.1375148	total: 378ms	remaining: 1.57s
194:	learn: 17534.8005706	total: 380ms	remaining: 1.

373:	learn: 12777.9175831	total: 708ms	remaining: 1.18s
374:	learn: 12759.3686963	total: 710ms	remaining: 1.18s
375:	learn: 12737.4437951	total: 711ms	remaining: 1.18s
376:	learn: 12701.7706887	total: 713ms	remaining: 1.18s
377:	learn: 12684.9730200	total: 715ms	remaining: 1.18s
378:	learn: 12672.0873149	total: 717ms	remaining: 1.17s
379:	learn: 12661.3335258	total: 719ms	remaining: 1.17s
380:	learn: 12636.9573045	total: 721ms	remaining: 1.17s
381:	learn: 12617.8979920	total: 723ms	remaining: 1.17s
382:	learn: 12598.9360603	total: 725ms	remaining: 1.17s
383:	learn: 12579.6943065	total: 727ms	remaining: 1.17s
384:	learn: 12571.7778802	total: 728ms	remaining: 1.16s
385:	learn: 12551.9084956	total: 730ms	remaining: 1.16s
386:	learn: 12523.5557659	total: 732ms	remaining: 1.16s
387:	learn: 12505.7080975	total: 734ms	remaining: 1.16s
388:	learn: 12484.8600921	total: 736ms	remaining: 1.16s
389:	learn: 12466.5902297	total: 737ms	remaining: 1.15s
390:	learn: 12437.3698920	total: 739ms	remaining

568:	learn: 9493.1493877	total: 1.06s	remaining: 807ms
569:	learn: 9469.5838737	total: 1.07s	remaining: 806ms
570:	learn: 9450.6400668	total: 1.07s	remaining: 804ms
571:	learn: 9448.0774635	total: 1.07s	remaining: 802ms
572:	learn: 9432.9568556	total: 1.07s	remaining: 800ms
573:	learn: 9426.1562767	total: 1.07s	remaining: 798ms
574:	learn: 9408.3747604	total: 1.08s	remaining: 796ms
575:	learn: 9402.3402321	total: 1.08s	remaining: 794ms
576:	learn: 9390.7191735	total: 1.08s	remaining: 792ms
577:	learn: 9372.5563562	total: 1.08s	remaining: 790ms
578:	learn: 9355.7386028	total: 1.08s	remaining: 789ms
579:	learn: 9350.4602550	total: 1.09s	remaining: 787ms
580:	learn: 9334.5979395	total: 1.09s	remaining: 785ms
581:	learn: 9318.7110463	total: 1.09s	remaining: 783ms
582:	learn: 9305.5216054	total: 1.09s	remaining: 781ms
583:	learn: 9291.6256017	total: 1.09s	remaining: 779ms
584:	learn: 9287.0128694	total: 1.09s	remaining: 777ms
585:	learn: 9276.0182696	total: 1.1s	remaining: 775ms
586:	learn:

759:	learn: 7320.4346645	total: 1.43s	remaining: 450ms
760:	learn: 7310.9559283	total: 1.43s	remaining: 448ms
761:	learn: 7306.3163937	total: 1.43s	remaining: 447ms
762:	learn: 7295.7702528	total: 1.43s	remaining: 445ms
763:	learn: 7287.8197936	total: 1.43s	remaining: 443ms
764:	learn: 7279.6547355	total: 1.44s	remaining: 441ms
765:	learn: 7270.6215867	total: 1.44s	remaining: 439ms
766:	learn: 7260.9115226	total: 1.44s	remaining: 437ms
767:	learn: 7249.6358201	total: 1.44s	remaining: 435ms
768:	learn: 7237.9759066	total: 1.44s	remaining: 433ms
769:	learn: 7232.2476470	total: 1.44s	remaining: 432ms
770:	learn: 7220.5609115	total: 1.45s	remaining: 430ms
771:	learn: 7213.0647055	total: 1.45s	remaining: 428ms
772:	learn: 7202.0870393	total: 1.45s	remaining: 426ms
773:	learn: 7197.4978076	total: 1.45s	remaining: 424ms
774:	learn: 7189.5567467	total: 1.45s	remaining: 422ms
775:	learn: 7178.4962311	total: 1.46s	remaining: 420ms
776:	learn: 7174.2196163	total: 1.46s	remaining: 418ms
777:	learn

953:	learn: 5693.4998604	total: 1.78s	remaining: 86ms
954:	learn: 5688.4437861	total: 1.79s	remaining: 84.2ms
955:	learn: 5682.6590010	total: 1.79s	remaining: 82.3ms
956:	learn: 5676.1318162	total: 1.79s	remaining: 80.4ms
957:	learn: 5668.2879894	total: 1.79s	remaining: 78.5ms
958:	learn: 5665.6517102	total: 1.79s	remaining: 76.7ms
959:	learn: 5659.0630897	total: 1.79s	remaining: 74.8ms
960:	learn: 5653.2536654	total: 1.8s	remaining: 72.9ms
961:	learn: 5644.8575288	total: 1.8s	remaining: 71.1ms
962:	learn: 5633.2620635	total: 1.8s	remaining: 69.2ms
963:	learn: 5627.6774794	total: 1.8s	remaining: 67.3ms
964:	learn: 5617.3086484	total: 1.8s	remaining: 65.5ms
965:	learn: 5611.8041701	total: 1.81s	remaining: 63.6ms
966:	learn: 5605.3449434	total: 1.81s	remaining: 61.7ms
967:	learn: 5598.3224444	total: 1.81s	remaining: 59.8ms
968:	learn: 5592.8652721	total: 1.81s	remaining: 58ms
969:	learn: 5585.9768878	total: 1.81s	remaining: 56.1ms
970:	learn: 5578.3227786	total: 1.81s	remaining: 54.2ms
9

177:	learn: 18239.2629779	total: 346ms	remaining: 1.6s
178:	learn: 18208.6536067	total: 348ms	remaining: 1.59s
179:	learn: 18164.6277609	total: 350ms	remaining: 1.59s
180:	learn: 18123.0983749	total: 351ms	remaining: 1.59s
181:	learn: 18092.5992831	total: 353ms	remaining: 1.59s
182:	learn: 18044.4155007	total: 355ms	remaining: 1.59s
183:	learn: 18001.1844912	total: 357ms	remaining: 1.58s
184:	learn: 17960.4824322	total: 359ms	remaining: 1.58s
185:	learn: 17911.9516709	total: 361ms	remaining: 1.58s
186:	learn: 17865.5630975	total: 363ms	remaining: 1.58s
187:	learn: 17824.9192841	total: 365ms	remaining: 1.58s
188:	learn: 17800.8888035	total: 367ms	remaining: 1.58s
189:	learn: 17757.7067431	total: 369ms	remaining: 1.57s
190:	learn: 17714.1811100	total: 371ms	remaining: 1.57s
191:	learn: 17681.8300486	total: 373ms	remaining: 1.57s
192:	learn: 17648.9044043	total: 375ms	remaining: 1.57s
193:	learn: 17599.1412764	total: 376ms	remaining: 1.56s
194:	learn: 17573.2972577	total: 378ms	remaining:

374:	learn: 12575.6055012	total: 706ms	remaining: 1.18s
375:	learn: 12548.4966096	total: 708ms	remaining: 1.17s
376:	learn: 12532.0017480	total: 709ms	remaining: 1.17s
377:	learn: 12504.8181799	total: 711ms	remaining: 1.17s
378:	learn: 12494.6300067	total: 713ms	remaining: 1.17s
379:	learn: 12460.9176178	total: 715ms	remaining: 1.17s
380:	learn: 12450.5152313	total: 717ms	remaining: 1.16s
381:	learn: 12429.7524394	total: 719ms	remaining: 1.16s
382:	learn: 12408.8267572	total: 721ms	remaining: 1.16s
383:	learn: 12388.0051050	total: 722ms	remaining: 1.16s
384:	learn: 12357.0486678	total: 724ms	remaining: 1.16s
385:	learn: 12338.5253820	total: 726ms	remaining: 1.16s
386:	learn: 12327.2858831	total: 728ms	remaining: 1.15s
387:	learn: 12303.8567245	total: 730ms	remaining: 1.15s
388:	learn: 12270.3108980	total: 732ms	remaining: 1.15s
389:	learn: 12243.3587019	total: 733ms	remaining: 1.15s
390:	learn: 12230.8309707	total: 735ms	remaining: 1.15s
391:	learn: 12207.1856881	total: 737ms	remaining

571:	learn: 9313.1451314	total: 1.06s	remaining: 797ms
572:	learn: 9299.5540266	total: 1.07s	remaining: 795ms
573:	learn: 9286.2448852	total: 1.07s	remaining: 794ms
574:	learn: 9273.1312808	total: 1.07s	remaining: 792ms
575:	learn: 9259.6632839	total: 1.07s	remaining: 790ms
576:	learn: 9243.5433689	total: 1.07s	remaining: 788ms
577:	learn: 9226.8291423	total: 1.08s	remaining: 786ms
578:	learn: 9211.7801640	total: 1.08s	remaining: 784ms
579:	learn: 9200.2391982	total: 1.08s	remaining: 782ms
580:	learn: 9183.6303176	total: 1.08s	remaining: 781ms
581:	learn: 9163.3872335	total: 1.08s	remaining: 779ms
582:	learn: 9148.2726843	total: 1.09s	remaining: 777ms
583:	learn: 9130.1263027	total: 1.09s	remaining: 775ms
584:	learn: 9116.6805929	total: 1.09s	remaining: 773ms
585:	learn: 9099.0835384	total: 1.09s	remaining: 771ms
586:	learn: 9089.9908184	total: 1.09s	remaining: 769ms
587:	learn: 9073.2154663	total: 1.09s	remaining: 768ms
588:	learn: 9056.0449708	total: 1.1s	remaining: 766ms
589:	learn:

764:	learn: 7158.8846513	total: 1.43s	remaining: 438ms
765:	learn: 7149.6901711	total: 1.43s	remaining: 436ms
766:	learn: 7141.4526795	total: 1.43s	remaining: 434ms
767:	learn: 7131.3264242	total: 1.43s	remaining: 432ms
768:	learn: 7122.2237432	total: 1.43s	remaining: 430ms
769:	learn: 7115.0019749	total: 1.43s	remaining: 428ms
770:	learn: 7105.7420525	total: 1.44s	remaining: 427ms
771:	learn: 7096.0963611	total: 1.44s	remaining: 425ms
772:	learn: 7082.4056723	total: 1.44s	remaining: 423ms
773:	learn: 7070.1214754	total: 1.44s	remaining: 421ms
774:	learn: 7060.8578796	total: 1.44s	remaining: 419ms
775:	learn: 7054.8111790	total: 1.45s	remaining: 417ms
776:	learn: 7046.0050692	total: 1.45s	remaining: 416ms
777:	learn: 7036.3615641	total: 1.45s	remaining: 414ms
778:	learn: 7023.4528745	total: 1.45s	remaining: 412ms
779:	learn: 7011.2859439	total: 1.45s	remaining: 410ms
780:	learn: 6997.2274670	total: 1.46s	remaining: 408ms
781:	learn: 6984.3570587	total: 1.46s	remaining: 406ms
782:	learn

954:	learn: 5648.5166418	total: 1.78s	remaining: 84ms
955:	learn: 5640.3679343	total: 1.78s	remaining: 82.2ms
956:	learn: 5630.5019257	total: 1.79s	remaining: 80.3ms
957:	learn: 5629.2485009	total: 1.79s	remaining: 78.4ms
958:	learn: 5618.1790671	total: 1.79s	remaining: 76.6ms
959:	learn: 5615.3181894	total: 1.79s	remaining: 74.7ms
960:	learn: 5614.1356856	total: 1.79s	remaining: 72.8ms
961:	learn: 5609.6696426	total: 1.8s	remaining: 71ms
962:	learn: 5602.3864805	total: 1.8s	remaining: 69.1ms
963:	learn: 5592.0893143	total: 1.8s	remaining: 67.2ms
964:	learn: 5587.2628782	total: 1.8s	remaining: 65.4ms
965:	learn: 5578.7315278	total: 1.8s	remaining: 63.5ms
966:	learn: 5571.3450363	total: 1.8s	remaining: 61.6ms
967:	learn: 5567.3090568	total: 1.81s	remaining: 59.8ms
968:	learn: 5558.0822930	total: 1.81s	remaining: 57.9ms
969:	learn: 5548.4977794	total: 1.81s	remaining: 56ms
970:	learn: 5542.5723892	total: 1.81s	remaining: 54.2ms
971:	learn: 5541.5248375	total: 1.81s	remaining: 52.3ms
972:

184:	learn: 18127.7523252	total: 356ms	remaining: 1.57s
185:	learn: 18072.2829670	total: 358ms	remaining: 1.57s
186:	learn: 18022.4211912	total: 360ms	remaining: 1.56s
187:	learn: 17968.6777056	total: 362ms	remaining: 1.56s
188:	learn: 17930.5895887	total: 364ms	remaining: 1.56s
189:	learn: 17876.6230315	total: 366ms	remaining: 1.56s
190:	learn: 17839.8574588	total: 368ms	remaining: 1.56s
191:	learn: 17804.2881415	total: 370ms	remaining: 1.55s
192:	learn: 17776.8852009	total: 372ms	remaining: 1.55s
193:	learn: 17731.1166399	total: 373ms	remaining: 1.55s
194:	learn: 17701.8176242	total: 375ms	remaining: 1.55s
195:	learn: 17660.5836440	total: 377ms	remaining: 1.55s
196:	learn: 17606.9616146	total: 379ms	remaining: 1.54s
197:	learn: 17563.9166396	total: 381ms	remaining: 1.54s
198:	learn: 17540.7605135	total: 383ms	remaining: 1.54s
199:	learn: 17522.8630194	total: 384ms	remaining: 1.54s
200:	learn: 17480.7442054	total: 386ms	remaining: 1.53s
201:	learn: 17443.7716047	total: 388ms	remaining

381:	learn: 12556.7021844	total: 716ms	remaining: 1.16s
382:	learn: 12527.1387302	total: 718ms	remaining: 1.16s
383:	learn: 12516.1764880	total: 720ms	remaining: 1.15s
384:	learn: 12493.2878973	total: 722ms	remaining: 1.15s
385:	learn: 12468.9572766	total: 724ms	remaining: 1.15s
386:	learn: 12451.9775964	total: 726ms	remaining: 1.15s
387:	learn: 12412.6795353	total: 728ms	remaining: 1.15s
388:	learn: 12380.7309441	total: 729ms	remaining: 1.15s
389:	learn: 12350.9335879	total: 731ms	remaining: 1.14s
390:	learn: 12347.3723993	total: 733ms	remaining: 1.14s
391:	learn: 12336.3201097	total: 735ms	remaining: 1.14s
392:	learn: 12321.8242805	total: 737ms	remaining: 1.14s
393:	learn: 12309.0492185	total: 739ms	remaining: 1.14s
394:	learn: 12268.3079777	total: 741ms	remaining: 1.13s
395:	learn: 12233.9191743	total: 742ms	remaining: 1.13s
396:	learn: 12219.2221014	total: 744ms	remaining: 1.13s
397:	learn: 12195.8628143	total: 746ms	remaining: 1.13s
398:	learn: 12187.5287633	total: 748ms	remaining

569:	learn: 9432.6922198	total: 1.08s	remaining: 812ms
570:	learn: 9424.5521711	total: 1.08s	remaining: 810ms
571:	learn: 9402.7757488	total: 1.08s	remaining: 808ms
572:	learn: 9389.9682273	total: 1.08s	remaining: 806ms
573:	learn: 9373.8145466	total: 1.08s	remaining: 805ms
574:	learn: 9362.8365647	total: 1.08s	remaining: 803ms
575:	learn: 9338.1531983	total: 1.09s	remaining: 801ms
576:	learn: 9329.6070547	total: 1.09s	remaining: 799ms
577:	learn: 9309.2237895	total: 1.09s	remaining: 797ms
578:	learn: 9288.6868371	total: 1.09s	remaining: 795ms
579:	learn: 9284.9022027	total: 1.09s	remaining: 793ms
580:	learn: 9264.6082038	total: 1.1s	remaining: 791ms
581:	learn: 9251.8115165	total: 1.1s	remaining: 789ms
582:	learn: 9245.9631373	total: 1.1s	remaining: 787ms
583:	learn: 9234.6116647	total: 1.1s	remaining: 785ms
584:	learn: 9223.4838858	total: 1.1s	remaining: 784ms
585:	learn: 9210.4271709	total: 1.11s	remaining: 782ms
586:	learn: 9202.7583998	total: 1.11s	remaining: 780ms
587:	learn: 918

765:	learn: 7264.9102600	total: 1.44s	remaining: 439ms
766:	learn: 7258.3164593	total: 1.44s	remaining: 437ms
767:	learn: 7250.5347154	total: 1.44s	remaining: 435ms
768:	learn: 7240.3239100	total: 1.44s	remaining: 433ms
769:	learn: 7230.6896114	total: 1.44s	remaining: 431ms
770:	learn: 7224.6085019	total: 1.45s	remaining: 429ms
771:	learn: 7218.6277670	total: 1.45s	remaining: 427ms
772:	learn: 7210.6420739	total: 1.45s	remaining: 425ms
773:	learn: 7201.7737257	total: 1.45s	remaining: 424ms
774:	learn: 7185.8750970	total: 1.45s	remaining: 422ms
775:	learn: 7167.5534280	total: 1.45s	remaining: 420ms
776:	learn: 7160.3633423	total: 1.46s	remaining: 418ms
777:	learn: 7151.9293905	total: 1.46s	remaining: 416ms
778:	learn: 7143.6086874	total: 1.46s	remaining: 414ms
779:	learn: 7132.7479340	total: 1.46s	remaining: 412ms
780:	learn: 7116.6774644	total: 1.46s	remaining: 411ms
781:	learn: 7111.9222469	total: 1.47s	remaining: 409ms
782:	learn: 7104.5092087	total: 1.47s	remaining: 407ms
783:	learn

960:	learn: 5634.5174286	total: 1.79s	remaining: 72.8ms
961:	learn: 5626.5779190	total: 1.8s	remaining: 71ms
962:	learn: 5618.7160495	total: 1.8s	remaining: 69.1ms
963:	learn: 5613.7317585	total: 1.8s	remaining: 67.2ms
964:	learn: 5600.0227873	total: 1.8s	remaining: 65.4ms
965:	learn: 5592.6807772	total: 1.8s	remaining: 63.5ms
966:	learn: 5586.3763441	total: 1.81s	remaining: 61.6ms
967:	learn: 5580.8325187	total: 1.81s	remaining: 59.8ms
968:	learn: 5574.0847915	total: 1.81s	remaining: 57.9ms
969:	learn: 5568.7715662	total: 1.81s	remaining: 56ms
970:	learn: 5561.5045866	total: 1.81s	remaining: 54.2ms
971:	learn: 5554.6932961	total: 1.81s	remaining: 52.3ms
972:	learn: 5549.1451088	total: 1.82s	remaining: 50.5ms
973:	learn: 5543.3580820	total: 1.82s	remaining: 48.6ms
974:	learn: 5532.7626523	total: 1.82s	remaining: 46.7ms
975:	learn: 5527.0050469	total: 1.82s	remaining: 44.8ms
976:	learn: 5520.3978677	total: 1.82s	remaining: 43ms
977:	learn: 5512.8572236	total: 1.83s	remaining: 41.1ms
978

179:	learn: 17950.1169793	total: 348ms	remaining: 1.58s
180:	learn: 17908.2869504	total: 349ms	remaining: 1.58s
181:	learn: 17866.4578487	total: 351ms	remaining: 1.58s
182:	learn: 17812.6213900	total: 353ms	remaining: 1.58s
183:	learn: 17777.1836798	total: 355ms	remaining: 1.57s
184:	learn: 17723.2706086	total: 357ms	remaining: 1.57s
185:	learn: 17680.5675753	total: 359ms	remaining: 1.57s
186:	learn: 17660.7213332	total: 361ms	remaining: 1.57s
187:	learn: 17621.9077363	total: 362ms	remaining: 1.56s
188:	learn: 17608.8104767	total: 364ms	remaining: 1.56s
189:	learn: 17564.9202249	total: 366ms	remaining: 1.56s
190:	learn: 17526.7452673	total: 368ms	remaining: 1.56s
191:	learn: 17494.5250283	total: 370ms	remaining: 1.56s
192:	learn: 17447.3665300	total: 372ms	remaining: 1.55s
193:	learn: 17394.7420872	total: 374ms	remaining: 1.55s
194:	learn: 17358.2786701	total: 375ms	remaining: 1.55s
195:	learn: 17304.2140958	total: 377ms	remaining: 1.55s
196:	learn: 17259.6230066	total: 379ms	remaining

375:	learn: 12542.8167121	total: 707ms	remaining: 1.17s
376:	learn: 12524.2954117	total: 709ms	remaining: 1.17s
377:	learn: 12513.7147243	total: 710ms	remaining: 1.17s
378:	learn: 12506.7616969	total: 712ms	remaining: 1.17s
379:	learn: 12496.4056938	total: 714ms	remaining: 1.17s
380:	learn: 12485.0614207	total: 716ms	remaining: 1.16s
381:	learn: 12474.1294913	total: 718ms	remaining: 1.16s
382:	learn: 12450.0820360	total: 720ms	remaining: 1.16s
383:	learn: 12428.9319523	total: 722ms	remaining: 1.16s
384:	learn: 12406.3648858	total: 724ms	remaining: 1.16s
385:	learn: 12395.1484520	total: 726ms	remaining: 1.15s
386:	learn: 12360.8871795	total: 728ms	remaining: 1.15s
387:	learn: 12347.0804775	total: 729ms	remaining: 1.15s
388:	learn: 12325.5925225	total: 731ms	remaining: 1.15s
389:	learn: 12307.1997850	total: 733ms	remaining: 1.15s
390:	learn: 12299.8036080	total: 735ms	remaining: 1.14s
391:	learn: 12288.6009393	total: 737ms	remaining: 1.14s
392:	learn: 12268.6083124	total: 738ms	remaining

571:	learn: 9571.3594950	total: 1.07s	remaining: 798ms
572:	learn: 9556.8732148	total: 1.07s	remaining: 797ms
573:	learn: 9543.7062095	total: 1.07s	remaining: 795ms
574:	learn: 9532.1201968	total: 1.07s	remaining: 793ms
575:	learn: 9517.1486521	total: 1.07s	remaining: 791ms
576:	learn: 9501.9120338	total: 1.08s	remaining: 789ms
577:	learn: 9492.1870386	total: 1.08s	remaining: 788ms
578:	learn: 9480.6262270	total: 1.08s	remaining: 786ms
579:	learn: 9472.4395790	total: 1.08s	remaining: 784ms
580:	learn: 9445.9015236	total: 1.08s	remaining: 782ms
581:	learn: 9431.5079238	total: 1.09s	remaining: 780ms
582:	learn: 9418.3528364	total: 1.09s	remaining: 778ms
583:	learn: 9406.2108272	total: 1.09s	remaining: 777ms
584:	learn: 9395.6114603	total: 1.09s	remaining: 775ms
585:	learn: 9384.3497397	total: 1.09s	remaining: 773ms
586:	learn: 9368.6410111	total: 1.09s	remaining: 771ms
587:	learn: 9353.9745739	total: 1.1s	remaining: 769ms
588:	learn: 9334.2855447	total: 1.1s	remaining: 767ms
589:	learn: 

766:	learn: 7469.1043229	total: 1.43s	remaining: 433ms
767:	learn: 7460.7466467	total: 1.43s	remaining: 432ms
768:	learn: 7454.6538106	total: 1.43s	remaining: 430ms
769:	learn: 7441.9584175	total: 1.43s	remaining: 428ms
770:	learn: 7430.9577355	total: 1.43s	remaining: 426ms
771:	learn: 7421.6433566	total: 1.44s	remaining: 424ms
772:	learn: 7408.8634015	total: 1.44s	remaining: 422ms
773:	learn: 7399.8109338	total: 1.44s	remaining: 420ms
774:	learn: 7394.9587665	total: 1.44s	remaining: 419ms
775:	learn: 7388.5265661	total: 1.44s	remaining: 417ms
776:	learn: 7376.7214704	total: 1.45s	remaining: 415ms
777:	learn: 7367.7898514	total: 1.45s	remaining: 413ms
778:	learn: 7359.5170065	total: 1.45s	remaining: 411ms
779:	learn: 7349.0820615	total: 1.45s	remaining: 409ms
780:	learn: 7338.9814642	total: 1.45s	remaining: 407ms
781:	learn: 7336.8667086	total: 1.45s	remaining: 406ms
782:	learn: 7328.7941706	total: 1.46s	remaining: 404ms
783:	learn: 7325.3803427	total: 1.46s	remaining: 402ms
784:	learn

961:	learn: 5916.5421957	total: 1.79s	remaining: 70.6ms
962:	learn: 5910.0331298	total: 1.79s	remaining: 68.7ms
963:	learn: 5898.4559991	total: 1.79s	remaining: 66.8ms
964:	learn: 5890.5284372	total: 1.79s	remaining: 65ms
965:	learn: 5883.0639691	total: 1.79s	remaining: 63.1ms
966:	learn: 5876.9368587	total: 1.79s	remaining: 61.3ms
967:	learn: 5876.2094116	total: 1.8s	remaining: 59.4ms
968:	learn: 5870.2390656	total: 1.8s	remaining: 57.6ms
969:	learn: 5862.6422969	total: 1.8s	remaining: 55.7ms
970:	learn: 5851.8665961	total: 1.8s	remaining: 53.9ms
971:	learn: 5844.5282888	total: 1.8s	remaining: 52ms
972:	learn: 5836.2608159	total: 1.81s	remaining: 50.1ms
973:	learn: 5828.4496624	total: 1.81s	remaining: 48.3ms
974:	learn: 5817.0161890	total: 1.81s	remaining: 46.4ms
975:	learn: 5807.9925971	total: 1.81s	remaining: 44.6ms
976:	learn: 5804.5044485	total: 1.81s	remaining: 42.7ms
977:	learn: 5798.7352970	total: 1.82s	remaining: 40.9ms
978:	learn: 5791.8777899	total: 1.82s	remaining: 39ms
979

180:	learn: 17631.6413677	total: 345ms	remaining: 1.56s
181:	learn: 17595.2754415	total: 347ms	remaining: 1.56s
182:	learn: 17565.4648217	total: 349ms	remaining: 1.56s
183:	learn: 17535.9281756	total: 350ms	remaining: 1.55s
184:	learn: 17485.3571835	total: 352ms	remaining: 1.55s
185:	learn: 17448.2290253	total: 354ms	remaining: 1.55s
186:	learn: 17406.3336973	total: 356ms	remaining: 1.55s
187:	learn: 17360.8226216	total: 358ms	remaining: 1.55s
188:	learn: 17327.6935262	total: 360ms	remaining: 1.54s
189:	learn: 17271.6732872	total: 362ms	remaining: 1.54s
190:	learn: 17249.4052102	total: 364ms	remaining: 1.54s
191:	learn: 17201.8871406	total: 366ms	remaining: 1.54s
192:	learn: 17172.1034111	total: 368ms	remaining: 1.54s
193:	learn: 17133.0803083	total: 369ms	remaining: 1.53s
194:	learn: 17100.3744264	total: 371ms	remaining: 1.53s
195:	learn: 17057.4347966	total: 373ms	remaining: 1.53s
196:	learn: 17012.3955725	total: 375ms	remaining: 1.53s
197:	learn: 16994.1188483	total: 377ms	remaining

378:	learn: 11977.8099725	total: 705ms	remaining: 1.15s
379:	learn: 11970.5600682	total: 707ms	remaining: 1.15s
380:	learn: 11951.2441279	total: 709ms	remaining: 1.15s
381:	learn: 11946.0973868	total: 710ms	remaining: 1.15s
382:	learn: 11933.3870436	total: 712ms	remaining: 1.15s
383:	learn: 11911.6813239	total: 714ms	remaining: 1.15s
384:	learn: 11889.6195123	total: 716ms	remaining: 1.14s
385:	learn: 11875.0898638	total: 718ms	remaining: 1.14s
386:	learn: 11847.9321157	total: 720ms	remaining: 1.14s
387:	learn: 11833.0636379	total: 722ms	remaining: 1.14s
388:	learn: 11805.7522721	total: 724ms	remaining: 1.14s
389:	learn: 11782.9089847	total: 727ms	remaining: 1.14s
390:	learn: 11759.1998333	total: 729ms	remaining: 1.14s
391:	learn: 11735.3780642	total: 731ms	remaining: 1.13s
392:	learn: 11715.6663997	total: 733ms	remaining: 1.13s
393:	learn: 11703.3996940	total: 734ms	remaining: 1.13s
394:	learn: 11688.9060717	total: 736ms	remaining: 1.13s
395:	learn: 11665.4255345	total: 738ms	remaining

572:	learn: 8842.2287487	total: 1.06s	remaining: 794ms
573:	learn: 8830.2802207	total: 1.07s	remaining: 792ms
574:	learn: 8822.7716002	total: 1.07s	remaining: 790ms
575:	learn: 8817.0128605	total: 1.07s	remaining: 789ms
576:	learn: 8805.0155001	total: 1.07s	remaining: 787ms
577:	learn: 8797.4964162	total: 1.07s	remaining: 785ms
578:	learn: 8786.7178637	total: 1.08s	remaining: 783ms
579:	learn: 8783.8625683	total: 1.08s	remaining: 781ms
580:	learn: 8769.0163644	total: 1.08s	remaining: 780ms
581:	learn: 8766.7041661	total: 1.08s	remaining: 778ms
582:	learn: 8753.8201316	total: 1.08s	remaining: 776ms
583:	learn: 8746.7329978	total: 1.09s	remaining: 774ms
584:	learn: 8733.7608402	total: 1.09s	remaining: 772ms
585:	learn: 8722.6810226	total: 1.09s	remaining: 770ms
586:	learn: 8714.0613368	total: 1.09s	remaining: 769ms
587:	learn: 8710.3822554	total: 1.09s	remaining: 767ms
588:	learn: 8702.7190850	total: 1.1s	remaining: 765ms
589:	learn: 8693.2640695	total: 1.1s	remaining: 764ms
590:	learn: 

764:	learn: 6896.8522957	total: 1.42s	remaining: 437ms
765:	learn: 6884.8659975	total: 1.43s	remaining: 436ms
766:	learn: 6877.2799520	total: 1.43s	remaining: 434ms
767:	learn: 6871.0768136	total: 1.43s	remaining: 432ms
768:	learn: 6865.5041223	total: 1.43s	remaining: 430ms
769:	learn: 6859.9252474	total: 1.43s	remaining: 428ms
770:	learn: 6850.5889380	total: 1.44s	remaining: 426ms
771:	learn: 6839.4856708	total: 1.44s	remaining: 424ms
772:	learn: 6833.3798019	total: 1.44s	remaining: 423ms
773:	learn: 6817.9859034	total: 1.44s	remaining: 421ms
774:	learn: 6805.0072144	total: 1.44s	remaining: 419ms
775:	learn: 6799.4308572	total: 1.44s	remaining: 417ms
776:	learn: 6789.8633289	total: 1.45s	remaining: 415ms
777:	learn: 6778.6631727	total: 1.45s	remaining: 413ms
778:	learn: 6773.7951676	total: 1.45s	remaining: 411ms
779:	learn: 6767.5622732	total: 1.45s	remaining: 410ms
780:	learn: 6758.4013374	total: 1.45s	remaining: 408ms
781:	learn: 6749.6271120	total: 1.46s	remaining: 406ms
782:	learn

961:	learn: 5496.4635126	total: 1.78s	remaining: 70.5ms
962:	learn: 5489.9748258	total: 1.79s	remaining: 68.7ms
963:	learn: 5481.1354873	total: 1.79s	remaining: 66.8ms
964:	learn: 5475.3925109	total: 1.79s	remaining: 65ms
965:	learn: 5468.7527069	total: 1.79s	remaining: 63.1ms
966:	learn: 5462.3687642	total: 1.79s	remaining: 61.3ms
967:	learn: 5453.5699375	total: 1.8s	remaining: 59.4ms
968:	learn: 5442.3076726	total: 1.8s	remaining: 57.5ms
969:	learn: 5440.4726868	total: 1.8s	remaining: 55.7ms
970:	learn: 5433.4917577	total: 1.8s	remaining: 53.8ms
971:	learn: 5426.3899699	total: 1.8s	remaining: 52ms
972:	learn: 5421.8134969	total: 1.81s	remaining: 50.1ms
973:	learn: 5411.6774486	total: 1.81s	remaining: 48.3ms
974:	learn: 5408.1559963	total: 1.81s	remaining: 46.4ms
975:	learn: 5401.9217787	total: 1.81s	remaining: 44.6ms
976:	learn: 5393.5732442	total: 1.81s	remaining: 42.7ms
977:	learn: 5389.1593809	total: 1.81s	remaining: 40.8ms
978:	learn: 5381.5452130	total: 1.82s	remaining: 39ms
979

176:	learn: 18265.6303472	total: 359ms	remaining: 1.67s
177:	learn: 18224.0839575	total: 361ms	remaining: 1.67s
178:	learn: 18172.6952613	total: 363ms	remaining: 1.66s
179:	learn: 18133.7872884	total: 365ms	remaining: 1.66s
180:	learn: 18094.0166713	total: 367ms	remaining: 1.66s
181:	learn: 18055.1707493	total: 369ms	remaining: 1.66s
182:	learn: 18030.5498940	total: 370ms	remaining: 1.65s
183:	learn: 17994.1353600	total: 372ms	remaining: 1.65s
184:	learn: 17957.0500028	total: 374ms	remaining: 1.65s
185:	learn: 17923.2856366	total: 376ms	remaining: 1.65s
186:	learn: 17877.1233203	total: 378ms	remaining: 1.64s
187:	learn: 17834.6713182	total: 380ms	remaining: 1.64s
188:	learn: 17779.3410239	total: 382ms	remaining: 1.64s
189:	learn: 17743.4216483	total: 383ms	remaining: 1.63s
190:	learn: 17688.8448546	total: 385ms	remaining: 1.63s
191:	learn: 17662.7138782	total: 387ms	remaining: 1.63s
192:	learn: 17625.7243391	total: 389ms	remaining: 1.63s
193:	learn: 17576.6013593	total: 391ms	remaining

368:	learn: 12841.3328593	total: 717ms	remaining: 1.23s
369:	learn: 12814.2927775	total: 719ms	remaining: 1.22s
370:	learn: 12787.3666553	total: 721ms	remaining: 1.22s
371:	learn: 12766.5847433	total: 723ms	remaining: 1.22s
372:	learn: 12745.9540385	total: 725ms	remaining: 1.22s
373:	learn: 12722.6303248	total: 727ms	remaining: 1.22s
374:	learn: 12709.5754698	total: 728ms	remaining: 1.21s
375:	learn: 12688.0868080	total: 730ms	remaining: 1.21s
376:	learn: 12673.2314078	total: 732ms	remaining: 1.21s
377:	learn: 12655.0883558	total: 734ms	remaining: 1.21s
378:	learn: 12650.1466129	total: 736ms	remaining: 1.21s
379:	learn: 12636.6108009	total: 738ms	remaining: 1.2s
380:	learn: 12621.3076868	total: 740ms	remaining: 1.2s
381:	learn: 12594.0403916	total: 742ms	remaining: 1.2s
382:	learn: 12581.9565652	total: 743ms	remaining: 1.2s
383:	learn: 12564.5978257	total: 745ms	remaining: 1.2s
384:	learn: 12547.5502962	total: 747ms	remaining: 1.19s
385:	learn: 12519.8938564	total: 749ms	remaining: 1.1

565:	learn: 9725.8936254	total: 1.08s	remaining: 826ms
566:	learn: 9714.2589955	total: 1.08s	remaining: 824ms
567:	learn: 9708.2236873	total: 1.08s	remaining: 822ms
568:	learn: 9693.8688735	total: 1.08s	remaining: 820ms
569:	learn: 9687.5114027	total: 1.08s	remaining: 818ms
570:	learn: 9668.2289787	total: 1.08s	remaining: 816ms
571:	learn: 9649.8508605	total: 1.09s	remaining: 814ms
572:	learn: 9646.2788366	total: 1.09s	remaining: 812ms
573:	learn: 9625.5198046	total: 1.09s	remaining: 810ms
574:	learn: 9613.0487079	total: 1.09s	remaining: 808ms
575:	learn: 9607.6428358	total: 1.09s	remaining: 806ms
576:	learn: 9594.2427156	total: 1.1s	remaining: 804ms
577:	learn: 9583.4006731	total: 1.1s	remaining: 802ms
578:	learn: 9576.5522256	total: 1.1s	remaining: 801ms
579:	learn: 9566.7915622	total: 1.1s	remaining: 799ms
580:	learn: 9555.1605549	total: 1.1s	remaining: 797ms
581:	learn: 9533.9914374	total: 1.11s	remaining: 795ms
582:	learn: 9520.0203984	total: 1.11s	remaining: 793ms
583:	learn: 950

761:	learn: 7648.7841802	total: 1.44s	remaining: 449ms
762:	learn: 7638.5451466	total: 1.44s	remaining: 447ms
763:	learn: 7633.6224102	total: 1.44s	remaining: 445ms
764:	learn: 7628.5332474	total: 1.44s	remaining: 443ms
765:	learn: 7621.2103114	total: 1.44s	remaining: 441ms
766:	learn: 7610.0846877	total: 1.45s	remaining: 439ms
767:	learn: 7601.5284647	total: 1.45s	remaining: 438ms
768:	learn: 7588.2899282	total: 1.45s	remaining: 436ms
769:	learn: 7581.5067024	total: 1.45s	remaining: 434ms
770:	learn: 7573.7336827	total: 1.45s	remaining: 432ms
771:	learn: 7562.7009587	total: 1.46s	remaining: 430ms
772:	learn: 7551.0308127	total: 1.46s	remaining: 428ms
773:	learn: 7547.2498890	total: 1.46s	remaining: 426ms
774:	learn: 7541.4399525	total: 1.46s	remaining: 424ms
775:	learn: 7536.6808575	total: 1.46s	remaining: 422ms
776:	learn: 7530.5215600	total: 1.46s	remaining: 420ms
777:	learn: 7518.0122887	total: 1.47s	remaining: 419ms
778:	learn: 7510.0950889	total: 1.47s	remaining: 417ms
779:	learn

959:	learn: 6143.9533687	total: 1.8s	remaining: 74.9ms
960:	learn: 6139.0440350	total: 1.8s	remaining: 73ms
961:	learn: 6128.3936374	total: 1.8s	remaining: 71.2ms
962:	learn: 6117.0583058	total: 1.8s	remaining: 69.3ms
963:	learn: 6109.7577503	total: 1.8s	remaining: 67.4ms
964:	learn: 6103.1071035	total: 1.81s	remaining: 65.6ms
965:	learn: 6100.0066759	total: 1.81s	remaining: 63.7ms
966:	learn: 6098.5524311	total: 1.81s	remaining: 61.8ms
967:	learn: 6087.5745514	total: 1.81s	remaining: 59.9ms
968:	learn: 6081.6365395	total: 1.81s	remaining: 58.1ms
969:	learn: 6079.8397261	total: 1.82s	remaining: 56.2ms
970:	learn: 6076.6153836	total: 1.82s	remaining: 54.3ms
971:	learn: 6069.3967866	total: 1.82s	remaining: 52.5ms
972:	learn: 6059.6135679	total: 1.82s	remaining: 50.6ms
973:	learn: 6059.0951059	total: 1.82s	remaining: 48.7ms
974:	learn: 6052.8819038	total: 1.83s	remaining: 46.8ms
975:	learn: 6047.5852914	total: 1.83s	remaining: 45ms
976:	learn: 6042.0078030	total: 1.83s	remaining: 43.1ms
9

GridSearchCV(cv=10,
             estimator=Pipeline(steps=[('pipe',
                                        Pipeline(steps=[('preprocessor',
                                                         ColumnTransformer(transformers=[('cat',
                                                                                          Pipeline(steps=[('transformer_function',
                                                                                                           CategoricalTransformer()),
                                                                                                          ('imputer',
                                                                                                           SimpleImputer(strategy='most_frequent')),
                                                                                                          ('encoder',
                                                                                                           OneHotEncoder(drop='if_binary',
                                                                                                                         handle_unknown='ignore')),
                                                                                                          ('selectKbest',
                                                                                                           SelectKBest(k=200))]),
                                                                                          <sk...
                                                                                                           StandardScaler()),
                                                                                                          ('pca',
                                                                                                           PCA(n_components=0.85))]),
                                                                                          <sklearn.compose._column_transformer.make_column_selector object at 0x0000015B7EB2C700>)]))])),
                                       ('clf', ClfSwitcher())]),
             param_grid=[{'clf__estimator': [<catboost.core.CatBoostRegressor object at 0x0000015B7E14F130>],
                          'pipe__preprocessor__cat__selectKbest__k': [200],
                          'pipe__preprocessor__num__pca__n_components': [0.85]}])

In [ ]:
gs = pd.DataFrame(gscv.cv_results_)
# Display the top ranking default models
display(gs.sort_values(by='rank_test_score', ascending=True))

mean_fit_time  std_fit_time  mean_score_time  std_score_time  \
0       2.286331      0.046238          0.01495        0.000002   

                                param_clf__estimator  \
0  <catboost.core.CatBoostRegressor object at 0x0...   

  param_pipe__preprocessor__cat__selectKbest__k  \
0                                           200   

  param_pipe__preprocessor__num__pca__n_components  \
0                                             0.85   

                                              params  split0_test_score  \
0  {'clf__estimator': <catboost.core.CatBoostRegr...           0.922741   

   split1_test_score  split2_test_score  split3_test_score  split4_test_score  \
0           0.905083           0.936919           0.864517           0.887816   

   split5_test_score  split6_test_score  split7_test_score  split8_test_score  \
0           0.898914           0.905562           0.916477           0.821915   

   split9_test_score  mean_test_score  std_test_score  rank_test_score  
0           0.902645         0.896259        0.030954                1

In [ ]:
gs.params[0]

{'clf__estimator': <catboost.core.CatBoostRegressor at 0x15b7e14f130>,
 'pipe__preprocessor__cat__selectKbest__k': 200,
 'pipe__preprocessor__num__pca__n_components': 0.85}

In [ ]:
best = gscv.best_estimator_
y_pred = best.predict(X_test)

mse = mean_squared_error(y_test, y_pred)
round(np.sqrt(mse),4)

5576.6415

cat_pca85_k200: 5208.2218
- Bath per bed, drop ['PoolQC','Fence','MiscFeature','FireplaceQu']

cat_pca85_k200: 5576.6415
- Bath per bed, summer_sale, drop ['PoolQC','Fence','MiscFeature','FireplaceQu']

cat_pca75_k201: 5545.9846

In [ ]:
pred = pd.DataFrame(y_pred, columns=['SalePrice'])
pred['Id'] = [i+1461 for i in range(0,len(pred))]
pred = pred[['Id','SalePrice']]
pred.head()

pred.to_csv('./submissions/Pipeline.csv', index=False)

### Base CatBoost

### Tuning CatBoost

# Other Code

### Model Selection

### Ensemble model

### Model selection gridsearchcv